test-cloud/test-cloud-remote/src/test/java/org/kie/cloud/integrationtests/integration/ProcessInstanceMigrationExternalDatabaseIntegrationTest.java

```diff
@@ -0,0 +1,210 @@
+        migrationDefinition.setProcessInstanceIds(Arrays.asList(instancesIds.get(0)));
+        migrationDefinition.setExecution(new Execution().setType(Execution.ExecutionType.SYNC));
+
+        HttpPost post = preparePost("/migrations");
+
+        StringWriter writer = new StringWriter();
+        mapper.writeValue(writer, migrationDefinition);
+        post.setEntity(new StringEntity(writer.toString()));
+        HttpResponse r = client.execute(post);
+        Assertions.assertThat(r.getStatusLine().getStatusCode()).isEqualTo(HttpStatus.SC_OK);
+        StringReader reader = new StringReader(EntityUtils.toString(r.getEntity()));
+        Migration migration = mapper.readValue(reader, Migration.class);
+        Assertions.assertThat(migration).isNotNull();
+        //Assertions.assertThat(migration.getDefinition().getRequester()).isEqualTo("kermit");
+        Assertions.assertThat(migration.getDefinition().getRequester()).isEqualTo("ANONYMOUS");
+        return migration;
+    }
+
+    private Plan createPlan() throws IOException, JAXBException {
+        Plan plan = new Plan();
+        plan.setSource(new ProcessRef().setContainerId(OLD_CONTAINER_ID).setProcessId(PROCESS_ID));
+        plan.setTarget(new ProcessRef().setContainerId(NEW_CONTAINER_ID).setProcessId(PROCESS_ID));
+
+        HttpPost post = preparePost("/plans");
+
+        StringWriter writer = new StringWriter();
+        mapper.writeValue(writer, plan);
+        post.setEntity(new StringEntity(writer.toString()));
+        HttpResponse r = client.execute(post);
+        Assertions.assertThat(r.getStatusLine().getStatusCode()).isEqualTo(HttpStatus.SC_OK);
+        StringReader reader = new StringReader(EntityUtils.toString(r.getEntity()));
+        return mapper.readValue(reader, Plan.class);
+    }
+
+    private HttpPost preparePost(String path) {
+        HttpPost post = new HttpPost(deploymentScenario.getProcessMigrationDeployment().getUrl()+"/rest" + path);
+        post.setHeader(HttpHeaders.ACCEPT, MediaType.APPLICATION_JSON);
+        post.setHeader(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON);
+        return post;
+    }
+
+    private void startProcesses() {
+        instancesIds.add(processServicesClient.startProcess(OLD_CONTAINER_ID, PROCESS_ID));
+        instancesIds.add(processServicesClient.startProcess(OLD_CONTAINER_ID, PROCESS_ID));
+    }
+
+
+    private CredentialsProvider getBasicAuth() {
+        CredentialsProvider provider = new BasicCredentialsProvider();
+        //provider.setCredentials(AuthScope.ANY, new UsernamePasswordCredentials(pimUsername, pimPassword));
+        provider.setCredentials(AuthScope.ANY, new UsernamePasswordCredentials("yoda", "usetheforce123@"));
+        return provider;
+    }
+ 
+    private static boolean isExternalDatabaseAllocated() {
+        if (DeploymentConstants.getDatabaseDriver().isEmpty()) {
+            return false;
+        } else {
+            return true;
+        }
```

Comment:
this should not be only:

{code}
return !DeploymentConstants.getDatabaseDriver().isEmpty();
{code}